solutions/script-component-ad/pages/_app.tsx

```diff
@@ -0,0 +1,38 @@
+import type { AppProps } from 'next/app'
+import type { LayoutProps } from '@vercel/examples-ui/layout'
+
+import { getLayout } from '@vercel/examples-ui'
+
+import '@vercel/examples-ui/globals.css'
+import Script from 'next/script'
+
+function App({ Component, pageProps }: AppProps) {
+  const Layout = getLayout<LayoutProps>(Component)
+
+  return (
+    <Layout
+      title="Script Component with ads"
+      path="solutions/script-component-ad"
+    >
+      <Script
+        strategy="afterInteractive"
+        id="script-component-ad"
+        src="https://securepubads.g.doubleclick.net/tag/js/gpt.js"
+      />
+
+      <Script id="define-slot" strategy="afterInteractive">{`
```

Comment:
We should use the `onLoad` prop in here, toggle some state and use that to trigger the script load instead of the useTimeout from above